src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -139,7 +139,7 @@ internal static class DiagnosticHelper
         {
             if (additionalUnnecessaryLocations.IsEmpty)
             {
-                return Create(descriptor, location, effectiveSeverity, additionalLocations, ImmutableDictionary<string, string?>.Empty, messageArgs);
+                return Create(descriptor, location, effectiveSeverity, additionalLocations, properties, messageArgs);
```

Comment:
The codefix relies on properties to know whether to include a diagnostic in FixAll operation or not. Not passing properties in this code path caused all diagnostics to be included in fix all.